tests/integration/deliver_grant_funding/test_routes.py

```diff
@@ -650,3 +643,152 @@ def test_edit_question_post(authenticated_platform_admin_client, factories, db_s
+        sess["grant_setup"] = {}
+
+    response = authenticated_platform_admin_client.get(url_for("deliver_grant_funding.grant_setup_name"))
+    assert response.status_code == 200
+    soup = BeautifulSoup(response.data, "html.parser")
+    assert soup.h1.text.strip() == "What is the name of this grant?"
+
+
+def test_grant_setup_name_post(authenticated_platform_admin_client):
+    with authenticated_platform_admin_client.session_transaction() as sess:
+        sess["grant_setup"] = {}
+
+    name_form = GrantNameForm(name="Test Grant Name")
+    response = authenticated_platform_admin_client.post(
+        url_for("deliver_grant_funding.grant_setup_name"), data=name_form.data, follow_redirects=False
+    )
+    assert response.status_code == 302
+    assert response.location == url_for("deliver_grant_funding.grant_setup_description")
+
+
+def test_grant_setup_description_get_with_session(authenticated_platform_admin_client):
+    with authenticated_platform_admin_client.session_transaction() as sess:
+        sess["grant_setup"] = {}
+
+    response = authenticated_platform_admin_client.get(url_for("deliver_grant_funding.grant_setup_description"))
+    assert response.status_code == 200
+    soup = BeautifulSoup(response.data, "html.parser")
+    assert soup.h1.text.strip() == "What is the main purpose of this grant?"
+
+
+def test_grant_setup_description_post_too_long(authenticated_platform_admin_client):
+    with authenticated_platform_admin_client.session_transaction() as sess:
+        sess["grant_setup"] = {}
+
+    long_description = " ".join(["word"] * 201)
+    desc_form = GrantDescriptionForm(description=long_description)
+    response = authenticated_platform_admin_client.post(
+        url_for("deliver_grant_funding.grant_setup_description"), data=desc_form.data, follow_redirects=False
+    )
+    assert response.status_code == 200
+    soup = BeautifulSoup(response.data, "html.parser")
+    assert soup.h2.text.strip() == "There is a problem"
+    assert len(soup.find_all("a", href="#description")) == 1
+    assert "Description must be 200 words or fewer" in soup.find_all("a", href="#description")[0].text.strip()
+
+
+def test_grant_setup_description_post_valid(authenticated_platform_admin_client):
+    with authenticated_platform_admin_client.session_transaction() as sess:
+        sess["grant_setup"] = {}
+
+    desc_form = GrantDescriptionForm(description="A valid description under 200 words.")
+    response = authenticated_platform_admin_client.post(
+        url_for("deliver_grant_funding.grant_setup_description"), data=desc_form.data, follow_redirects=False
+    )
+    assert response.status_code == 302
+    assert response.location == url_for("deliver_grant_funding.grant_setup_contact")
+
+
+def test_grant_setup_contact_get_with_session(authenticated_platform_admin_client):
+    with authenticated_platform_admin_client.session_transaction() as sess:
+        sess["grant_setup"] = {}
```

Comment:
Do we think it's worth updating these tests that set the session to `{}` to have representative data in them? I think useful to understand the flows as a developer reviewing the tests, but will leave to you on whether it's too much maintenance burden.